http-client/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java

```diff
@@ -355,6 +356,14 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
             uriVariables.forEach(queryParams::remove);
 
             request = HttpRequest.create(httpMethod, appendQuery(uri, queryParams), httpMethodName);
+
+            Stream<Argument> argumentStream = Arrays.stream(arguments);
+            Optional<Argument> basicAuthArgument = argumentStream.filter(arg -> arg.getType() == BasicAuth.class).findFirst();
+            basicAuthArgument.ifPresent(a -> {
+                final BasicAuth auth = (BasicAuth) paramMap.get(a.getName());
+                request.basicAuth(auth.getUsername(), auth.getPassword());
+            });
```

Comment:
Rather than creating a stream and filtering all the arguments can this be handled in the `for` loop that is already processing arguments? See https://github.com/micronaut-projects/micronaut-core/blob/e0ff1a5923ac524fe5eeb36611251155b7aa46a1/http-client/src/main/java/io/micronaut/http/client/interceptor/HttpClientIntroductionAdvice.java#L243

Reply:
I did check the ```for``` loop above that you are mentioning. But, I wanted to keep BasicAuth instance final and it will have to accessible after ```request``` creation happens for us to be able to set the value on it. Can keep it non-final, assign value in ```for``` loop and then set it after ```request``` creation.

Reply:
In that case, `BasicAuth` instance will have to be kept uninitialised and checked for presence of a value in it again or presence of the argument before adding it to the ```request```.